File: printer/printer.js
/*
문제 출처 : https://programmers.co.kr/learn/courses/30/lessons/42587
입출력 예
priorities	            location	         return
[2, 1, 3, 2]	           2	               1
[1, 1, 9, 1, 1, 1]	       0	               5
*/

const printer = (priorities, location) => {
  let count = 1;
  let target_index = location;
  let first;

  while (priorities.length > 0) {
    first = priorities.shift();
    if (priorities.some((value, idx) => value > first)) {
      priorities.push(first);
    } else {
      if (target_index === 0) {
        break;
      } else {
        count++;
      }
    }
    if (target_index === 0) {
      target_index = priorities.length - 1;
    } else {
      target_index--;
    }
  }
  return count;
};

export default printer;

File: Programmers/printer/printer.py
def printer(priorities, location):
    count = 0
    m = max(priorities)

    while True:
        first = priorities.pop(0)
        if first == m:
            count += 1
            if location == 0:
                break
            else:
                location -= 1
            m = max(priorities)
        else:
            priorities.append(first)
            if location == 0:
                location = len(priorities) - 1
            else:
                location -= 1

    print(count)
    return count


# printer([2, 1, 3, 2], 2)
# printer([1, 1, 9, 1, 1, 1], 0)
